internal/controller/storageclient_controller.go

```diff
@@ -54,20 +65,32 @@ const (
 	storageClientFinalizer            = "storageclient.ocs.openshift.io"
 	storageClientDefaultAnnotationKey = "ocs.openshift.io/storageclient.default"
 
+	vgscClusterIDIndexName = "index:volumeGroupSnapshotContentCSIDriver"
+
 	csvPrefix = "ocs-client-operator"
 
 	VolumeGroupSnapshotClassCrdName = "volumegroupsnapshotclasses.groupsnapshot.storage.k8s.io"
 )
 
+var (
+	crdsToWatch = []string{VolumeGroupSnapshotClassCrdName}
+	csiDrivers  = []string{templates.RBDDriverName, templates.CephFsDriverName}
+)
+
 // StorageClientReconciler reconciles a StorageClient object
 type StorageClientReconciler struct {
 	client.Client
 	Scheme            *runtime.Scheme
 	OperatorNamespace string
+
+	cache            cache.Cache
+	controller       controller.Controller
+	mu               sync.RWMutex
+	crdsBeingWatched map[string]bool
```

Comment:
Why use this and not a sync map to simplify the code and maintenance?

Reply:
done.